src/coreclr/tools/aot/ObjWriter/debugInfo/dwarf/dwarfGen.cpp

```diff
@@ -284,6 +284,42 @@ static int GetDwarfFpRegNum(Triple::ArchType ArchType)
   }
 }
 
+static int GetRegOpSize(int DwarfRegNum) {
+  if (DwarfRegNum <= 31) {
+    return 1;
+  }
+  else if (DwarfRegNum < 128) {
+    return 2;
+  }
+  else if (DwarfRegNum < 16384) {
+    return 3;
+  }
+  else {
+    assert(false && "Too big register number");
+    return 0;
+  }
+}
+
+static void EmitBreg(MCObjectStreamer* Streamer, int DwarfRegNum) {
+  if (DwarfRegNum <= 31) {
+    Streamer->EmitIntValue(DwarfRegNum + dwarf::DW_OP_breg0, 1);
+  }
+  else {
+    Streamer->EmitIntValue(dwarf::DW_OP_bregx, 1);
```

Comment:
According to the dwarf manual: "The DW_OP_bregx operation has two operands: a register which is specified by an unsigned LEB128 number, followed by a signed LEB128 offset.". It would be nice to emit the offset as part of this function instead of emitting it in the calling code.